src/main/kotlin/io/kweb/plugins/staticFiles/StaticFilesPlugin.kt

```diff
@@ -1,29 +1,53 @@
 package io.kweb.plugins.staticFiles
 
 import io.ktor.http.content.files
+import io.ktor.http.content.resources
 import io.ktor.http.content.static
 import io.ktor.http.content.staticRootFolder
+import io.ktor.routing.Route
 import io.ktor.routing.Routing
 import io.kweb.plugins.KwebPlugin
+import io.kweb.plugins.jqueryCore.jqueryCore
 import java.io.File
+import java.net.URL
 
 /**
  * @author rpanic
  *
  * This Plugin serves static files to be used in the frontend
  *
  * @property rootFolder The root folder, where the static assets are saved
+ * @property resourceFolder For serving resources, the path to the folder which will be served
  * @property servedRoute The route where these assets are being served
  */
-class StaticFilesPlugin(private val rootFolder: File, private val servedRoute: String = "assets") : KwebPlugin(){
+class StaticFilesPlugin private constructor(private val servedRoute: String = "assets") : KwebPlugin() {
 
-    override fun decorate(startHead: StringBuilder, endHead: StringBuilder){}
+    private lateinit var datasource: (Route) -> Unit
+
+    constructor(rootFolder: File, servedRoute: String = "assets") : this(servedRoute) {
+        datasource = {
+            it.staticRootFolder = rootFolder
+        }
+    }
+
+    constructor(resourceFolder: ResourceFolder, servedRoute: String = "assets") : this(servedRoute) {
+        datasource = {
+            it.resources(resourceFolder.resourceFolder)
+        }
+    }
+
+    override fun decorate(startHead: StringBuilder, endHead: StringBuilder) {}
 
     override fun appServerConfigurator(routeHandler: Routing) {
-        routeHandler.static(servedRoute){
-            staticRootFolder = rootFolder
+        routeHandler.static(servedRoute) {
+            datasource(this)
             files(".")
         }
     }
 
 }
+
+internal const val internalStaticFilePath = "kweb_native_assets"
+internal val internalStaticFilesPlugin = StaticFilesPlugin(ResourceFolder("io/kweb/assets"), servedRoute = internalStaticFilePath)
```

Comment:
Would it make more sense to use the plugin's package?  ie.

io/kweb/assets -> io/kweb/plugins/fomanticUI

Reply:
I changed it so that no .class files get exposed and the structure is as clean as possible :)